// plugin-welcome/src/api/models/ControllersBooking.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Project API Booking
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersBooking
 */
export interface ControllersBooking {
    /**
     * 
     * @type {string}
     * @memberof ControllersBooking
     */
    added?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersBooking
     */
    operationroom?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersBooking
     */
    patient?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersBooking
     */
    user?: number;
}

export function ControllersBookingFromJSON(json: any): ControllersBooking {
    return ControllersBookingFromJSONTyped(json, false);
}

export function ControllersBookingFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersBooking {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'added': !exists(json, 'added') ? undefined : json['added'],
        'operationroom': !exists(json, 'operationroom') ? undefined : json['operationroom'],
        'patient': !exists(json, 'patient') ? undefined : json['patient'],
        'user': !exists(json, 'user') ? undefined : json['user'],
    };
}

export function ControllersBookingToJSON(value?: ControllersBooking | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'added': value.added,
        'operationroom': value.operationroom,
        'patient': value.patient,
        'user': value.user,
    };
}
