// plugin-welcome/src/api/models/EntPatientEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Project API Booking
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBooking,
    EntBookingFromJSON,
    EntBookingFromJSONTyped,
    EntBookingToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPatientEdges
 */
export interface EntPatientEdges {
    /**
     * PatientID holds the value of the patient_id edge.
     * @type {Array<EntBooking>}
     * @memberof EntPatientEdges
     */
    patientID?: Array<EntBooking>;
}

export function EntPatientEdgesFromJSON(json: any): EntPatientEdges {
    return EntPatientEdgesFromJSONTyped(json, false);
}

export function EntPatientEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPatientEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'patientID': !exists(json, 'patientID') ? undefined : ((json['patientID'] as Array<any>).map(EntBookingFromJSON)),
    };
}

export function EntPatientEdgesToJSON(value?: EntPatientEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'patientID': value.patientID === undefined ? undefined : ((value.patientID as Array<any>).map(EntBookingToJSON)),
    };
}
