// plugin-welcome/src/api/models/EntUser.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Project API Booking
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntUserEdges,
    EntUserEdgesFromJSON,
    EntUserEdgesFromJSONTyped,
    EntUserEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntUser
 */
export interface EntUser {
    /**
     * DoctorEmail holds the value of the "doctor_email" field.
     * @type {string}
     * @memberof EntUser
     */
    doctorEmail?: string;
    /**
     * DoctorName holds the value of the "doctor_name" field.
     * @type {string}
     * @memberof EntUser
     */
    doctorName?: string;
    /**
     * 
     * @type {EntUserEdges}
     * @memberof EntUser
     */
    edges?: EntUserEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntUser
     */
    id?: number;
}

export function EntUserFromJSON(json: any): EntUser {
    return EntUserFromJSONTyped(json, false);
}

export function EntUserFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntUser {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'doctorEmail': !exists(json, 'doctor_email') ? undefined : json['doctor_email'],
        'doctorName': !exists(json, 'doctor_name') ? undefined : json['doctor_name'],
        'edges': !exists(json, 'edges') ? undefined : EntUserEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntUserToJSON(value?: EntUser | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'doctor_email': value.doctorEmail,
        'doctor_name': value.doctorName,
        'edges': EntUserEdgesToJSON(value.edges),
        'id': value.id,
    };
}
